packages/pigeon/lib/objc_generator.dart

```diff
@@ -710,11 +761,16 @@ static NSDictionary<NSString *, id> *wrapResult(id result, FlutterError *error)
         } else {
           indent.writeln(
               '$resultName.${field.name} = ${_dictGetter(classNames, 'dict', field, options.prefix)};');
-          indent.write(
-              'if ((NSNull *)$resultName.${field.name} == [NSNull null]) ');
-          indent.scoped('{', '}', () {
-            indent.writeln('$resultName.${field.name} = nil;');
-          });
+          if (field.type.isNullable) {
+            indent.write(
+                'if ((NSNull *)$resultName.${field.name} == [NSNull null]) ');
```

Comment:
That's not a convention, that's how you store conceptually null values in a dictionary. I'm not asking why there are `NSNull`s in the dictionary though; there's no type violation there because the values in an NSDictionary are, by default (as in that code), `NSObject`s, and `NSNull` is an `NSObject`. I'm asking why `field`, whose type is presumably not `NSObject` has an `NSNull` stored in it.

The proper way to extract a possibly-null value from a dictionary into a typed variable is to check it *before* making what would be a type-violating assignment. I.e.:
```
NSString *foo = myDict['fooKey'] == [NSNull null] ? nil : myDict['fooKey'];
if (foo != nil) {
  ...
}
```

is perfectly valid code. But
```
NSString *foo = myDict['fooKey'];
if (foo != nil && foo != [NSNull null]) {
  ..
}
```

is not, because there is a codepath there where the thing `foo` is pointing to is **an unrelated type**. (And when I say it's not valid code, I mean that it's code we should not write/generate because it violates the type contract; I know it will technically work because it's a dynamic runtime.)

Reply:
I see, the reason I've done the later is to avoid multiple lookups and to avoid introducing more naming.  When you are doing codegen anytime you create a name (variable, function, class) you run a risk of colliding, so I avoid it.  Other metaprogramming systems solve this with hygienic macros or something like common-lisp's gensym.  We don't have those available to us so I sometimes do things to avoid introducing new symbols.

Reply:
Here is some generate code from that area:
```objc
// header
@interface SearchRequest : NSObject
/// `init` unavailable to enforce nonnull fields, see the `make` class method.
- (instancetype)init NS_UNAVAILABLE;
+ (instancetype)makeWithQuery:(NSString *)query;
@property(nonatomic, copy) NSString * query;
@end

// source
@interface SearchRequest ()
+ (SearchRequest *)fromMap:(NSDictionary *)dict;
- (NSDictionary *)toMap;
@end

@implementation SearchRequest
+ (instancetype)makeWithQuery:(NSString *)query {
  SearchRequest* pigeonResult = [[SearchRequest alloc] init];
  pigeonResult.query = query;
  return pigeonResult;
}
+ (SearchRequest *)fromMap:(NSDictionary *)dict {
  SearchRequest *pigeonResult = [[SearchRequest alloc] init];
  pigeonResult.query = dict[@"query"];
  NSAssert((NSNull *)pigeonResult.query != [NSNull null], @"");
  return pigeonResult;
}
- (NSDictionary *)toMap {
  return [NSDictionary dictionaryWithObjectsAndKeys:(self.query ? self.query : [NSNull null]), @"query", nil];
}
@end
```

So, any variable name we chose there for a local variable could collide with with the field names.  We can mitigate that by adding a "pigeon" prefix to the variable names.  Like I said over vc, I just wanted to avoid any issue.

Reply:
We should definitely fix that using prefixed variables; we shouldn't generated code that's abusing the type system. It can be a follow-up since it's pre-existing.